transport/src/main/java/io/netty/channel/ManualIoEventLoop.java

```diff
@@ -148,28 +149,71 @@ public Ticker ticker() {
         return ticker;
     }
 
-    private int runAllTasks() {
+    /**
+     * Poll all tasks from the task queue and run them via {@link Runnable#run()} method.
+     * <br>
+     * The behaviour of this method is the same as {@link #runNonIoTasks(long)}
+     * with {@code timeoutNanos} set {@code <= 0}.
+     */
+    public int runNonIoTasks() {
+        return runAllTasks(-1, true);
+    }
+
+    /**
+     * Poll all tasks from the task queue and run them via {@link Runnable#run()} method.  This method stops running
+     * the tasks in the task queue and returns if it ran longer than {@code timeoutNanos}.<br>
```

Comment:
We could rephrase the javadoc like "Poll and run tasks from the task queue, until the task queue is empty or the given deadline (if positive) is exceeded."

I think that'd be clearer.